src/Connection.php

```diff
@@ -327,13 +345,15 @@ protected function getDefaultPostProcessor()
     /** @inheritdoc */
     protected function getDefaultQueryGrammar()
     {
-        return new Query\Grammar();
+        // Argument added in Laravel 12
+        return new Query\Grammar($this);
     }
 
     /** @inheritdoc */
     protected function getDefaultSchemaGrammar()
     {
-        return new Schema\Grammar();
+        // Argument added in Laravel 12
+        return new Schema\Grammar($this);
```

Comment:
The constructor was introduced by https://github.com/laravel/framework/commit/c78fa3d0684206f721b9a3b76e8c596aa8a08cd0#diff-13a4975d935d982e846853d4328df60c7ef014b5a4010441277b2ef4de31ed2aR27